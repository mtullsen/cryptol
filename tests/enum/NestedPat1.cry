enum M a = N | J a

f x = case x of
        J (a#b) -> a + b
        N -> zero











