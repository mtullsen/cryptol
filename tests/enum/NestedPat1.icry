:load NestedPat1.cry
f N
f (J 17) : [8]
