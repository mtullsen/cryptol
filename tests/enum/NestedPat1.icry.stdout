Loading module Cryptol
Loading module Cryptol
Loading module Main
Showing a specific instance of polymorphic result:
  * Using 'Integer' for 2nd type argument of 'Main::f'
  * Using '0' for 1st type argument of 'Main::f'
[]
0x11
